projects/cds/src/profiletag/services/profile-tag-push-events.service.ts

```diff
@@ -0,0 +1,150 @@
+import { Injectable } from '@angular/core';
+import {
+  ActiveCartService,
+  EventService,
+  PersonalizationContextService,
+} from '@spartacus/core';
+import {
+  CartPageVisited,
+  CategoryPageVisited,
+  HomePageVisited,
+  KeywordSearchPageVisited,
+  OrderConfirmationPageVisited,
+  PageVisited,
+  ProductDetailsPageVisited,
+} from 'projects/core/src/routing/event/routing.events';
+import { merge, Observable, of } from 'rxjs';
+import { map, skipWhile, withLatestFrom } from 'rxjs/operators';
+import {
+  CartChangedPushEvent,
+  CartViewPushEvent,
+  CategoryViewPushEvent,
+  HomePageViewPushEvent,
+  KeywordSearchPushEvent,
+  OrderConfirmationPushEvent,
+  PageViewPushEvent,
+  ProductViewPushEvent,
+  ProfileTagPushEvent,
+} from '../model/profile-tag.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ProfileTagPushEventsService {
```

Comment:
Let's add some brief API comments on the class and method level.

Reply:
Done.